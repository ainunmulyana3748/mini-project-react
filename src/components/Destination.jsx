import React from "react";

const Destination = () => {
  const images = [
    {
      id: 1,
      url: "https://akcdn.detik.net.id/visual/2019/07/03/66048cd8-30b6-42a3-9e3c-1cb5f655a720_169.jpeg?w=1200",
      title: "Tour Jepang",
    },
    {
      id: 2,
      url: "https://a.travel-assets.com/findyours-php/viewfinder/images/res70/474000/474240-Left-Bank-Paris.jpg",
      title: "Paris Experience",
    },
    {
      id: 3,
      url: "https://www.cimbniaga.co.id/content/dam/cimb/inspirasi/jungfraujoch.webp",
      title: "Swiss Journey",
    },
    {
      id: 4,
      url: "https://cdn.kimkim.com/files/a/content_articles/featured_photos/a09710eeb604a66276bc74d304e6d801c14ced98/big-cb0fc75914a607f6b9065f8a5b191a09.jpg",
      title: "Iceland Adventure",
    },
  ];

  return (
    <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-4 gap-6">
      {images.map((item) => (
        <div
          key={item.id}
          className="bg-white rounded-2xl overflow-hidden shadow-lg cursor-pointer hover:shadow-xl hover:scale-105 transform transition duration-300 ease-in-out"
        >
          <img
            src={item.url}
            alt={item.title}
            className="h-48 w-full object-cover"
          />
          <div className="p-4">
            <h2 className="text-lg font-semibold text-gray-800">
              {item.title}
            </h2>
            <p className="text-sm text-gray-500 mt-1">
              Temukan pengalaman baru di destinasi impianmu.
            </p>
          </div>
        </div>
      ))}
    </div>
  );
};

export default Destination;
