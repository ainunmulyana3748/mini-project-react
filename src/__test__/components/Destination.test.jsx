import { render, screen } from "@testing-library/react";
import Destination from "../../components/Destination";

describe("Destination Component", () => {
  test("render semua destinasi dengan benar", () => {
    render(<Destination />);
    expect(screen.getByText("Tour Jepang")).toBeInTheDocument();
    expect(screen.getByText("Paris Experience")).toBeInTheDocument();
    expect(screen.getByText("Swiss Journey")).toBeInTheDocument();
    expect(screen.getByText("Iceland Adventure")).toBeInTheDocument();
    // Cek gambar
    expect(screen.getByAltText("Tour Jepang")).toBeInTheDocument();
    expect(screen.getByAltText("Paris Experience")).toBeInTheDocument();
    expect(screen.getByAltText("Swiss Journey")).toBeInTheDocument();
    expect(screen.getByAltText("Iceland Adventure")).toBeInTheDocument();
  });

  test("snapshot destination", () => {
    const { container } = render(<Destination />);
    expect(container).toMatchSnapshot();
  });
});
