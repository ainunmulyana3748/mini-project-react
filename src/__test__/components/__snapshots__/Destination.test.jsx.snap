// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Destination Component snapshot destination 1`] = `
<div>
  <div
    class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-4 gap-6"
  >
    <div
      class="bg-white rounded-2xl overflow-hidden shadow-lg cursor-pointer hover:shadow-xl hover:scale-105 transform transition duration-300 ease-in-out"
    >
      <img
        alt="Tour Jepang"
        class="h-48 w-full object-cover"
        src="https://akcdn.detik.net.id/visual/2019/07/03/66048cd8-30b6-42a3-9e3c-1cb5f655a720_169.jpeg?w=1200"
      />
      <div
        class="p-4"
      >
        <h2
          class="text-lg font-semibold text-gray-800"
        >
          Tour Jepang
        </h2>
        <p
          class="text-sm text-gray-500 mt-1"
        >
          Temukan pengalaman baru di destinasi impianmu.
        </p>
      </div>
    </div>
    <div
      class="bg-white rounded-2xl overflow-hidden shadow-lg cursor-pointer hover:shadow-xl hover:scale-105 transform transition duration-300 ease-in-out"
    >
      <img
        alt="Paris Experience"
        class="h-48 w-full object-cover"
        src="https://a.travel-assets.com/findyours-php/viewfinder/images/res70/474000/474240-Left-Bank-Paris.jpg"
      />
      <div
        class="p-4"
      >
        <h2
          class="text-lg font-semibold text-gray-800"
        >
          Paris Experience
        </h2>
        <p
          class="text-sm text-gray-500 mt-1"
        >
          Temukan pengalaman baru di destinasi impianmu.
        </p>
      </div>
    </div>
    <div
      class="bg-white rounded-2xl overflow-hidden shadow-lg cursor-pointer hover:shadow-xl hover:scale-105 transform transition duration-300 ease-in-out"
    >
      <img
        alt="Swiss Journey"
        class="h-48 w-full object-cover"
        src="https://www.cimbniaga.co.id/content/dam/cimb/inspirasi/jungfraujoch.webp"
      />
      <div
        class="p-4"
      >
        <h2
          class="text-lg font-semibold text-gray-800"
        >
          Swiss Journey
        </h2>
        <p
          class="text-sm text-gray-500 mt-1"
        >
          Temukan pengalaman baru di destinasi impianmu.
        </p>
      </div>
    </div>
    <div
      class="bg-white rounded-2xl overflow-hidden shadow-lg cursor-pointer hover:shadow-xl hover:scale-105 transform transition duration-300 ease-in-out"
    >
      <img
        alt="Iceland Adventure"
        class="h-48 w-full object-cover"
        src="https://cdn.kimkim.com/files/a/content_articles/featured_photos/a09710eeb604a66276bc74d304e6d801c14ced98/big-cb0fc75914a607f6b9065f8a5b191a09.jpg"
      />
      <div
        class="p-4"
      >
        <h2
          class="text-lg font-semibold text-gray-800"
        >
          Iceland Adventure
        </h2>
        <p
          class="text-sm text-gray-500 mt-1"
        >
          Temukan pengalaman baru di destinasi impianmu.
        </p>
      </div>
    </div>
  </div>
</div>
`;
